test/plugins/scenarios/mux/main.go

```diff
@@ -54,12 +56,50 @@ func health(w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte("success"))
 }
 
+var upgrader = websocket.Upgrader{}
+
+func ws(w http.ResponseWriter, r *http.Request) {
+	c, err := upgrader.Upgrade(w, r, nil)
+	if err != nil {
+		log.Print("upgrade:", err)
+		return
+	}
+	defer c.Close()
+	for {
+		mt, message, err := c.ReadMessage()
+		if err != nil {
+			log.Println("read:", err)
+			break
+		}
+		log.Printf("recv: %s", message)
+		err = c.WriteMessage(mt, message)
+		if err != nil {
+			log.Println("write:", err)
+			break
+		}
+	}
+}
+
 func main() {
 	r := mux.NewRouter()
 	r.Path("/health").HandlerFunc(health)
 	r.Path("/consumer").HandlerFunc(consumer)
 	r.PathPrefix("/provider").Path("/{var}").HandlerFunc(provider)
+	r.Path("/ws").HandlerFunc(ws)
+
+	go func() {
+		time.Sleep(3 * time.Second)
 
+		u := url.URL{Scheme: "ws", Host: "localhost:8080", Path: "/ws"}
+		log.Printf("connecting to %s", u.String())
+		c, _, err := websocket.DefaultDialer.Dial(u.String(), nil)
+		if err != nil {
+			log.Fatal("dial:", err)
+		}
+		defer c.Close()
+		c.WriteMessage(websocket.TextMessage, []byte("hello from mux test"))
```

Comment:
Could you move this message to the provider handler? The provider handler is the entrypoint.

Reply:
Done.